### DISCUSSION TITLE
nanoff --listdevices crashing

### DISCUSSION THREAD

**Discussion by martinmcnally** [nanoframework/Home#1525] [General]
When i run nanoff --listdevices it just freezes and does nothing crashing PowerShell. 

![image](https://github.com/user-attachments/assets/6c5a761f-3295-493d-b5eb-2795e263a9a5)

I can see my device in device explorer and can connect using PUTTY every thing seems normal. I have updated VS, removed and reinstalled nanoff, erased the firmware cache and re flashed the device etc. 

Still shows No available COM port and crashing when running listdevices.

Any ideas?

Thanks
Martin

**Comment by alberk8** [comment]
I find that the nanoff --listports unreliable.  on my machine it still list com ports that is non existent.    The Device Manager is the only source of truth for com ports.   I only trust COM port as listed there.

**Reply by josesimoes** [reply]
`--listdevices` operation may run into issues with COM ports provided by Bluetooth devices and other virtual ports.
At this time, because there is no way to specify a list of port to exclude from the search, I would recommend using it with caution.

**Reply by josesimoes** [reply]
As for `--listcomports` it's a simple wrapper to System.IO.Ports.SerialPort.GetPortNames() API from .NET anything not working as expected it's up to the OS...

**Reply by martinmcnally** [reply]
Any ideas how I can get VS to see this device?

**Reply by alberk8** [reply]
What is make and model of ESP32 that you are using now?

**Comment by martinmcnally** [comment]
ESP32-DEVKITC-32EESP32-WROOM-32E 4MB FLASHHopefully I didn’t get the wrong one. MartinOn Aug 5, 2024, at 5:48 PM, alberk8 ***@***.***> wrote:﻿
What is make and model of ESP32 that you are using now?

**Reply by alberk8** [reply]
Try to flash firmware with this cli command

`nanoff --target ESP32_REV3 --update --serialport COM4 --masserase`

**Reply by martinmcnally** [reply]
Thanks for the advice I re-flashed it today but it still did not work. 

You can connect to it fine in putty but VS still cant see it.

![image](https://github.com/user-attachments/assets/9a62f493-5b83-44bc-8ca6-6c5aec06a32f)

![image](https://github.com/user-attachments/assets/70414c58-3108-4623-9d59-2b5c8268418d)

**Reply by alberk8** [reply]
From the screen capture it looks like the Firmware is loaded successfully.   Make sure the usb to serial driver is the latest.

**Reply by josesimoes** [reply]
Something is messing up with VS. Make sure that you don't have more than one instance of VS open, that other extensions aren't locking/using the COM port (Arduino, for example does that) and that you haven't incidentallyturned off the device watcher in Device Explorer window toolbar. Also that the COM port is not on the exclusion list, also in device explorer toolbar.

**Reply by martinmcnally** [reply]
Yea the driver is the latest or the best one according to windows update and the port is not in use. This is obviously a problem with Visual Studio. 

Do the Port Settings need changed?

![image](https://github.com/user-attachments/assets/27cbdd55-a407-4364-b23b-6907f1cb17e3)

**Reply by alberk8** [reply]
No, you do not need to change any port settings

**Reply by josesimoes** [reply]
@martinmcnally this is very odd... 
What is the VS version that you are using? What is the nanoFramework extension version? 
And can you please copy here the output of the activity log of the extension?

**Reply by josesimoes** [reply]
Also, from the print screen above is not clear what fw version is running there. 
Can you add here the output of the boot start that you see in putty?

**Reply by martinmcnally** [reply]
![image](https://github.com/user-attachments/assets/137345f4-ce81-4cfa-9b64-45889cf290ba)

![image](https://github.com/user-attachments/assets/9f36f973-dfe9-4165-aedc-67a4e04e0007)

**Reply by martinmcnally** [reply]
![image](https://github.com/user-attachments/assets/10fc36c2-f983-4537-9774-5d1f59d9da1c)

![image](https://github.com/user-attachments/assets/489360e1-fdbe-402f-8787-a0f026c4bd0b)

**Reply by alberk8** [reply]
What is the command that you use to flash the firmware?  Also if possible post a picture of the ESP32 board.

**Reply by josesimoes** [reply]
@martinmcnally this is what I'm getting right now on my VS setup (same versions as you have).

![image](https://github.com/user-attachments/assets/ef559ee7-af75-474a-a11f-d0425293b1c1)

If there is no mention to the COM port of the nanoFramework device you have connected, that's because the `SerialPort` couldn't be opened. That means that something else on your system has opened _that_ COM port and it's using it thus preventing the extension from querying the device...
 
Double check for other VS extensions or any other software using/searching COM ports. Arduino is one of those reportedly messing the search workflow.

**Reply by martinmcnally** [reply]
nanoff --target ESP32_REV3 --update --serialport COM1 --masserase

I switched it to COM1 to see if COM5 was the issue. 

Ill keep trying to see what is up however VS is the only program running on the machine and its been reinstalled. Cant imagine anything else is using the COM ports but ill keep digging. 

This the board (has one red LED on when plugged in). 
![ESP32-DEVKITC-32E](https://github.com/user-attachments/assets/22f1c33a-a5d1-4c44-9e8f-4c0cf45e0c67)

**Reply by josesimoes** [reply]
@martinmcnally after VS is running if you unplug and plug back the USB cable, following the USB sound, do you see any message on that verbose output of the extension?

**Reply by martinmcnally** [reply]
I bought a new board arrived today and connects just fine in VS. 

This one:
https://www.amazon.ca/dp/B0C9TGJRPH?ref=ppx_yo2ov_dt_b_product_details&th=1

Not sure what is up with the other one I bought from DigiKey but im glad this one is working.

Thanks for all the help. 

--listdevices even works now too
![image](https://github.com/user-attachments/assets/a861b060-b5e5-4fe8-abdd-1e7c733f6c73)

Weird that the other board could run the firmware but not connect to it or show as a real device.

**Reply by alberk8** [reply]
one possibility is the LDO is not providing enough power (that is the three legged IC with a big pad on the opposite side) below the ESP32 module).
